source/common/conn_pool/conn_pool_base.cc

```diff
@@ -425,18 +425,21 @@ void ConnPoolImplBase::checkForIdleAndCloseIdleConnsIfDraining() {
 
 void ConnPoolImplBase::onConnectionEvent(ActiveClient& client, absl::string_view failure_reason,
                                          Network::ConnectionEvent event) {
-  if (client.state() == ActiveClient::State::CONNECTING) {
-    ASSERT(connecting_stream_capacity_ >= client.effectiveConcurrentStreamLimit());
-    connecting_stream_capacity_ -= client.effectiveConcurrentStreamLimit();
-  }
+  if (event != Network::ConnectionEvent::ConnectedZeroRtt) {
```

Comment:
comment why we don't do these for zero rtt

alternately can we rework the if into 
if (CONNECTING && !ZeroRTT) { 
  capacity work
  if (timer) {
   reset
  }
}